common/flatpak-dir.c

```diff
@@ -14316,7 +14316,11 @@ flatpak_dir_check_installed_ref_missing_related_ref (FlatpakDir          *self,
                                                          cancellable, &local_error);
   if (remote_related_refs == NULL)
     {
-      g_warning ("Unable to get remote related refs for %s: %s", full_ref, local_error->message);
+      if (local_error != NULL)
+        g_warning ("Unable to get remote related refs for %s: %s", full_ref, local_error->message);
+      else
+        g_warning ("Unable to get remote related refs for %s", full_ref);
```

Comment:
If this is expected when the remote is the origin for a bundle, then it shouldn't be a warning. Warnings are for unexpected conditions according to the `g_warning` docs.

Reply:
I don't understand this crash, it seems to me that the only reason it would crash is due to your other change that makes it return NULL instead of an empty GPtrArray if the url is empty. If you revert that change, does it still crash?

Reply:
The crash was happening in `flatpak_dir_check_installed_ref_missing_related_ref` because :
```C

  remote_related_refs = flatpak_dir_find_remote_related (self, state, full_ref,
                                                         cancellable, &local_error);
  if (remote_related_refs == NULL)
    {
      g_warning ("Unable to get remote related refs for %s: %s", full_ref, local_error->message);
      return FALSE;
    }
```

`flatpak_dir_find_remote_related` was returning NULL(without error set) instead of empty non-array(which it should if remote's `url= `)  and then the local_error was de-referenced.

Updated the PR to address it specifically.